pages/[templateId].tsx

```diff
@@ -58,6 +104,64 @@ const MarketplaceTemplateDetail = (): JSX.Element => {
     }
   }, [templateId]);
 
+  useEffect(() => {
+    setPurchasingError('');
+    if (balanceAmount === 0) setIsBalanceInsufficient(true);
+  }, [currentUser, currentUserBalance]);
+
+  useEffect(() => {
+    templateAssets.forEach((asset) => {
+      if (asset.salePrice === lowestPrice) {
+        setSaleId(asset.saleId);
+      }
+    });
+  }, [templateAssets]);
```

Comment:
Noticed that `lowestPrice` isnt finished fetching before the `useEffect` runs. So the initial value isn't getting set. Should be an easy enough fix to add `lowestPrice` to the dependency list for the `useEffect`

Reply:
Added `lowestPrice` as a dependency!